clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -166,6 +169,26 @@ public ConsumerCoordinator(LogContext logContext,
         if (autoCommitEnabled)
             this.nextAutoCommitTimer = time.timer(autoCommitIntervalMs);
 
+        // select the rebalance protocol such that:
+        //   1. only consider protocols that are supported by all the assignors. If there is no common protocols supported
+        //      across all the assignors, throw an exception.
+        //   2. if there are multiple protocols that are commonly supported, select the one with the highest id (i.e. the
+        //      id number indicates how advanced is the protocol).
+        // we know there are at least one assignor in the list, no need to double check for NPE
+        List<RebalanceProtocol> supportedProtocols = new ArrayList<>(assignors.get(0).supportedProtocols());
+
+        for (PartitionAssignor assignor : assignors) {
+            supportedProtocols.retainAll(assignor.supportedProtocols());
+        }
+
+        if (supportedProtocols.isEmpty()) {
+            throw new IllegalArgumentException("Specified assignors do not have commonly supported rebalance protocol");
```

Comment:
Do we want to include assignor value here?

Reply:
Good point.